mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -91,8 +91,8 @@ private[classification] trait LogisticRegressionParams extends ProbabilisticClas
   @Since("2.1.0")
   final val family: Param[String] = new Param(this, "family",
     "The name of family which is a description of the label distribution to be used in the " +
-      s"model. Supported options: ${supportedFamilyNames.mkString(", ")}.",
-    ParamValidators.inArray[String](supportedFamilyNames))
+      s"model (case-insensitive). Supported options: ${supportedFamilyNames.mkString(", ")}.",
```

Comment:
maybe we can add an additional string param validators class then to the same params.scala file in ml folder?  There should be a generic function and the params.scala file seems to be the right place.